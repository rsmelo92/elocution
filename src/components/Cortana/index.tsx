import styles from './styles.module.css';

export default function Cortana() {
  return (
    <div className={styles.cortana}>
      <div className={styles.outer}>
        <div className={styles.inner}></div>
      </div>
    </div>
  )
}
