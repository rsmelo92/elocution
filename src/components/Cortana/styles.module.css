.cortana {
  position: relative;
}

.inner, .outer {
  border-radius: 50%;
  top: 0;
  left: 0;
}

.inner {
  background-color: var(--primary-color);
  opacity: 0.6;
  top: 10px;
  left: 10px;
  height: 20px;
  width: 20px;
  animation: inner 1.5s infinite ease-in-out alternate;
  cursor: pointer;
}

.outer {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: var(--secondary-color);
  opacity: 0.5;
  height: 40px;
  width: 40px;
  transform-origin: center;
  animation: outer 1.5s infinite ease-in-out alternate;
}

@keyframes inner {
  0% {
    transform: scale(1);
  }
  40% {
    transform: scale(1.5);
  }
  100% {
    transform: scale(1);
  }
}

@keyframes outer {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.5);
  }
  100% {
    transform: scale(1);
  }
}
