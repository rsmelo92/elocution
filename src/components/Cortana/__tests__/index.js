import { render } from '@testing-library/react';
import Cortana from '../';

describe('Cortana', () => {
  test('renders correctly', () => {
    const { container } = render(<Cortana />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
