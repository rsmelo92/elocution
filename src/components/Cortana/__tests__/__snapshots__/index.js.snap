// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cortana renders correctly 1`] = `
<div
  class="cortana"
>
  <div
    class="outer"
  >
    <div
      class="inner"
    />
  </div>
</div>
`;
